Watching for file changes with StatReloader
C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection1app\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection1app\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection1app\views.py changed, reloading.
Watching for file changes with StatReloader
Index page accessed
"GET /prefix/ HTTP/1.1" 200 40
Ошибка на странице about: division by zero
Traceback (most recent call last):
  File "C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection1app\views.py", line 15, in about
    result = 10 / 0
             ~~~^~~
ZeroDivisionError: division by zero
"GET /prefix/about/ HTTP/1.1" 200 31
C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lessonsofdjango\settings.py changed, reloading.
Watching for file changes with StatReloader
Index page accessed
"GET /prefix/ HTTP/1.1" 200 40
C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lessonsofdjango\settings.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lessonsofdjango\settings.py changed, reloading.
INFO 2023-09-25 16:41:30,818 autoreload 1180 13572 Watching for file changes with StatReloader
INFO 2023-09-25 16:41:53,811 views 1180 15156 Index page accessed
INFO 2023-09-25 16:41:53,812 basehttp 1180 15156 "GET /prefix/ HTTP/1.1" 200 40
INFO 2023-09-28 09:44:38,445 autoreload 9808 8896 Watching for file changes with StatReloader
INFO 2023-09-28 09:45:57,382 autoreload 3228 2340 Watching for file changes with StatReloader
INFO 2023-09-28 09:47:47,661 autoreload 14992 6760 Watching for file changes with StatReloader
INFO 2023-09-28 09:50:33,620 autoreload 7892 8440 Watching for file changes with StatReloader
INFO 2023-09-29 10:13:57,627 autoreload 3952 6104 Watching for file changes with StatReloader
WARNING 2023-09-29 10:13:59,696 log 3952 11952 Not Found: /
WARNING 2023-09-29 10:13:59,697 basehttp 3952 11952 "GET / HTTP/1.1" 404 2280
WARNING 2023-09-29 10:13:59,801 log 3952 11952 Not Found: /favicon.ico
WARNING 2023-09-29 10:13:59,801 basehttp 3952 11952 "GET /favicon.ico HTTP/1.1" 404 2331
WARNING 2023-09-29 10:14:08,721 log 3952 11952 Not Found: /lec3
WARNING 2023-09-29 10:14:08,722 basehttp 3952 11952 "GET /lec3 HTTP/1.1" 404 2310
WARNING 2023-09-29 10:14:16,093 log 3952 11952 Not Found: /lec3/
WARNING 2023-09-29 10:14:16,094 basehttp 3952 11952 "GET /lec3/ HTTP/1.1" 404 2565
INFO 2023-09-29 10:14:29,871 basehttp 3952 11952 "GET /lec3/hello HTTP/1.1" 301 0
INFO 2023-09-29 10:14:29,877 basehttp 3952 15668 "GET /lec3/hello/ HTTP/1.1" 200 29
WARNING 2023-09-29 10:14:35,737 log 3952 15668 Not Found: /lec3/hello2
WARNING 2023-09-29 10:14:35,737 basehttp 3952 15668 "GET /lec3/hello2 HTTP/1.1" 404 2583
INFO 2023-09-29 10:14:51,906 basehttp 3952 15668 "GET /lec3/hello/ HTTP/1.1" 200 29
WARNING 2023-09-29 10:14:58,060 log 3952 15668 Not Found: /lec3/hello2/
WARNING 2023-09-29 10:14:58,061 basehttp 3952 15668 "GET /lec3/hello2/ HTTP/1.1" 404 2586
ERROR 2023-09-29 10:15:14,093 log 3952 15668 Internal Server Error: /lec3/hello/2
Traceback (most recent call last):
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\views\generic\base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\views\generic\base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: HelloWiew.get() got an unexpected keyword argument 'name'
ERROR 2023-09-29 10:15:14,095 basehttp 3952 15668 "GET /lec3/hello/2 HTTP/1.1" 500 70149
INFO 2023-09-29 10:15:47,910 autoreload 3952 6104 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\urls.py changed, reloading.
INFO 2023-09-29 10:15:48,495 autoreload 10596 11016 Watching for file changes with StatReloader
INFO 2023-09-29 10:15:55,983 basehttp 10596 2448 "GET /lec3/hello2 HTTP/1.1" 301 0
INFO 2023-09-29 10:15:55,990 basehttp 10596 11216 "GET /lec3/hello2/ HTTP/1.1" 200 26
INFO 2023-09-29 10:21:36,998 autoreload 10596 11016 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\urls.py changed, reloading.
INFO 2023-09-29 10:21:37,617 autoreload 13016 11952 Watching for file changes with StatReloader
INFO 2023-09-29 10:26:06,383 autoreload 13016 11952 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:26:06,928 autoreload 13768 10032 Watching for file changes with StatReloader
INFO 2023-09-29 10:26:24,629 autoreload 13768 10032 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:26:25,248 autoreload 10492 16196 Watching for file changes with StatReloader
INFO 2023-09-29 10:28:10,864 autoreload 10492 16196 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:28:11,435 autoreload 13684 11552 Watching for file changes with StatReloader
INFO 2023-09-29 10:29:32,191 autoreload 13684 11552 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:29:32,823 autoreload 6500 9996 Watching for file changes with StatReloader
INFO 2023-09-29 10:29:47,402 autoreload 6500 9996 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:29:47,956 autoreload 2008 15956 Watching for file changes with StatReloader
INFO 2023-09-29 10:30:07,586 autoreload 2008 15956 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\urls.py changed, reloading.
INFO 2023-09-29 10:30:08,157 autoreload 9160 13912 Watching for file changes with StatReloader
INFO 2023-09-29 10:31:42,023 autoreload 9160 13912 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\urls.py changed, reloading.
INFO 2023-09-29 10:31:42,581 autoreload 10388 11452 Watching for file changes with StatReloader
INFO 2023-09-29 10:31:51,995 autoreload 10388 11452 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\urls.py changed, reloading.
INFO 2023-09-29 10:31:52,547 autoreload 11392 14788 Watching for file changes with StatReloader
INFO 2023-09-29 10:33:22,674 autoreload 2520 13888 Watching for file changes with StatReloader
INFO 2023-09-29 10:33:27,403 basehttp 2520 15612 "GET /lec3/posts/2022 HTTP/1.1" 301 0
INFO 2023-09-29 10:33:27,407 basehttp 2520 11432 "GET /lec3/posts/2022/ HTTP/1.1" 200 19
INFO 2023-09-29 10:33:53,986 autoreload 2520 13888 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:33:54,545 autoreload 3168 15452 Watching for file changes with StatReloader
INFO 2023-09-29 10:33:56,548 basehttp 3168 14828 "GET /lec3/posts/2022/ HTTP/1.1" 200 45
INFO 2023-09-29 10:34:02,411 basehttp 3168 14828 "GET /lec3/posts/2022/6 HTTP/1.1" 301 0
INFO 2023-09-29 10:34:02,416 basehttp 3168 7596 "GET /lec3/posts/2022/6/ HTTP/1.1" 200 47
INFO 2023-09-29 10:36:34,937 autoreload 3168 15452 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:36:35,728 autoreload 492 7024 Watching for file changes with StatReloader
INFO 2023-09-29 10:38:10,477 autoreload 492 7024 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:38:11,104 autoreload 10112 2224 Watching for file changes with StatReloader
WARNING 2023-09-29 10:38:14,678 log 10112 15584 Not Found: /lec3/posts/2022/6/гусеница
WARNING 2023-09-29 10:38:14,679 basehttp 10112 15584 "GET /lec3/posts/2022/6/%D0%B3%D1%83%D1%81%D0%B5%D0%BD%D0%B8%D1%86%D0%B0 HTTP/1.1" 404 3309
INFO 2023-09-29 10:38:25,513 basehttp 10112 15584 "GET /lec3/posts/2022/6/gusenica HTTP/1.1" 301 0
INFO 2023-09-29 10:38:25,519 basehttp 10112 15640 "GET /lec3/posts/2022/6/gusenica/ HTTP/1.1" 200 333
INFO 2023-09-29 10:44:21,245 autoreload 10112 2224 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:44:21,810 autoreload 8448 6104 Watching for file changes with StatReloader
INFO 2023-09-29 10:46:24,720 autoreload 8448 6104 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\urls.py changed, reloading.
INFO 2023-09-29 10:46:25,273 autoreload 1872 11452 Watching for file changes with StatReloader
INFO 2023-09-29 10:46:30,880 basehttp 1872 8452 "GET /lec3 HTTP/1.1" 301 0
INFO 2023-09-29 10:46:30,889 basehttp 1872 9188 "GET /lec3/ HTTP/1.1" 200 187
INFO 2023-09-29 10:53:21,026 autoreload 1872 11452 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:53:21,676 autoreload 5964 10976 Watching for file changes with StatReloader
INFO 2023-09-29 10:54:18,956 autoreload 5964 10976 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:54:19,543 autoreload 11016 13932 Watching for file changes with StatReloader
INFO 2023-09-29 10:55:21,819 autoreload 11016 13932 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:55:22,414 autoreload 8628 1580 Watching for file changes with StatReloader
INFO 2023-09-29 10:56:06,274 autoreload 8628 1580 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\urls.py changed, reloading.
INFO 2023-09-29 10:56:06,867 autoreload 9480 14396 Watching for file changes with StatReloader
INFO 2023-09-29 10:56:17,500 autoreload 9480 14396 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\urls.py changed, reloading.
INFO 2023-09-29 10:56:18,108 autoreload 10408 12224 Watching for file changes with StatReloader
ERROR 2023-09-29 10:56:23,827 log 10408 10136 Internal Server Error: /lec3/if
Traceback (most recent call last):
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\smartif.py", line 179, in translate_token
    op = OPERATORS[token]
         ~~~~~~~~~^^^^^^^
KeyError: '<=4'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\response.py", line 90, in rendered_content
    template = self.resolve_template(self.template_name)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\response.py", line 72, in resolve_template
    return select_template(template, using=self.using)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader.py", line 42, in select_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\engine.py", line 157, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 200, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\defaulttags.py", line 960, in do_if
    condition = TemplateIfParser(parser, bits).parse()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\defaulttags.py", line 885, in __init__
    super().__init__(*args, **kwargs)
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\smartif.py", line 170, in __init__
    mapped_tokens.append(self.translate_token(token))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\smartif.py", line 181, in translate_token
    return self.create_var(token)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\defaulttags.py", line 888, in create_var
    return TemplateLiteral(self.template_parser.compile_filter(value), value)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 600, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 703, in __init__
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: '<=4' from '<=4'
ERROR 2023-09-29 10:56:23,836 basehttp 10408 10136 "GET /lec3/if HTTP/1.1" 500 165132
INFO 2023-09-29 10:56:42,917 basehttp 10408 10136 "GET /lec3/if HTTP/1.1" 200 561
INFO 2023-09-29 10:56:51,964 autoreload 10408 12224 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:56:52,539 autoreload 15604 6308 Watching for file changes with StatReloader
INFO 2023-09-29 10:56:53,873 basehttp 15604 6064 "GET /lec3/if HTTP/1.1" 200 562
INFO 2023-09-29 10:57:03,407 autoreload 15604 6308 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 10:57:04,023 autoreload 10396 15724 Watching for file changes with StatReloader
INFO 2023-09-29 10:57:05,005 basehttp 10396 4232 "GET /lec3/if HTTP/1.1" 200 558
INFO 2023-09-29 11:33:41,059 autoreload 10396 15724 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py changed, reloading.
INFO 2023-09-29 11:33:41,695 autoreload 8576 6500 Watching for file changes with StatReloader
INFO 2023-09-29 11:34:15,033 autoreload 8576 6500 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\urls.py changed, reloading.
INFO 2023-09-29 11:34:15,641 autoreload 1920 8312 Watching for file changes with StatReloader
INFO 2023-09-29 11:34:19,887 basehttp 1920 88 "GET /lec3/for HTTP/1.1" 200 536
ERROR 2023-09-29 11:34:45,123 log 1920 88 Internal Server Error: /lec3/for
Traceback (most recent call last):
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py", line 59, in view_for
    return render(request, 'lection3app/for_template.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\engine.py", line 157, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 200, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\defaulttags.py", line 852, in do_for
    sequence = parser.compile_filter(bits[in_index + 1])
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 600, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 703, in __init__
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: '()' from 'my_dict.items()'
ERROR 2023-09-29 11:34:45,128 basehttp 1920 88 "GET /lec3/for HTTP/1.1" 500 145298
ERROR 2023-09-29 11:35:02,601 log 1920 88 Internal Server Error: /lec3/for
Traceback (most recent call last):
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py", line 59, in view_for
    return render(request, 'lection3app/for_template.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\engine.py", line 157, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 200, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\defaulttags.py", line 852, in do_for
    sequence = parser.compile_filter(bits[in_index + 1])
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 600, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 703, in __init__
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: '()' from 'my_dict.items()'
ERROR 2023-09-29 11:35:02,606 basehttp 1920 88 "GET /lec3/for HTTP/1.1" 500 145298
INFO 2023-09-29 11:35:14,358 basehttp 1920 88 "GET /lec3/for HTTP/1.1" 200 745
INFO 2023-09-29 12:05:51,144 basehttp 1920 4584 "GET /lec3/for HTTP/1.1" 200 285
INFO 2023-09-29 12:06:22,919 basehttp 1920 4584 "GET /lec3/for HTTP/1.1" 200 286
INFO 2023-09-29 12:06:42,457 basehttp 1920 4584 "GET /lec3/for HTTP/1.1" 200 289
INFO 2023-09-29 12:07:13,770 basehttp 1920 4584 "GET /lec3/for HTTP/1.1" 200 274
INFO 2023-09-29 12:07:26,848 basehttp 1920 4584 "GET /lec3/for HTTP/1.1" 200 760
INFO 2023-09-29 12:09:12,742 basehttp 1920 4584 "GET /lec3/if HTTP/1.1" 200 569
INFO 2023-09-29 12:09:21,208 basehttp 1920 4584 "GET /lec3/for HTTP/1.1" 200 760
INFO 2023-09-29 12:13:08,117 autoreload 1920 8312 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lessonsofdjango\settings.py changed, reloading.
INFO 2023-09-29 12:13:08,752 autoreload 13156 8668 Watching for file changes with StatReloader
ERROR 2023-09-29 12:15:47,559 log 13156 1864 Internal Server Error: /lec3/for
Traceback (most recent call last):
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader_tags.py", line 132, in render
    compiled_parent = self.get_parent(context)
                      ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader_tags.py", line 129, in get_parent
    return self.find_template(parent, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader_tags.py", line 107, in find_template
    template, origin = context.template.engine.find_template(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\engine.py", line 161, in find_template
    raise TemplateDoesNotExist(name, tried=tried)
django.template.exceptions.TemplateDoesNotExist: base.html

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\views.py", line 59, in view_for
    return render(request, 'lection3app/for_template.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\backends\django.py", line 63, in render
    reraise(exc, self.backend)
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\backends\django.py", line 84, in reraise
    raise new from exc
django.template.exceptions.TemplateDoesNotExist: base.html
ERROR 2023-09-29 12:15:47,566 basehttp 13156 1864 "GET /lec3/for HTTP/1.1" 500 130918
INFO 2023-09-29 12:17:04,891 autoreload 13156 8668 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lessonsofdjango\settings.py changed, reloading.
INFO 2023-09-29 12:17:05,530 autoreload 16180 13304 Watching for file changes with StatReloader
ERROR 2023-09-29 12:18:29,531 log 16180 6332 Internal Server Error: /lec3/if
Traceback (most recent call last):
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader_tags.py", line 132, in render
    compiled_parent = self.get_parent(context)
                      ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader_tags.py", line 129, in get_parent
    return self.find_template(parent, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader_tags.py", line 107, in find_template
    template, origin = context.template.engine.find_template(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\engine.py", line 161, in find_template
    raise TemplateDoesNotExist(name, tried=tried)
django.template.exceptions.TemplateDoesNotExist: base.html

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\backends\django.py", line 63, in render
    reraise(exc, self.backend)
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\backends\django.py", line 84, in reraise
    raise new from exc
django.template.exceptions.TemplateDoesNotExist: base.html
ERROR 2023-09-29 12:18:29,536 basehttp 16180 6332 "GET /lec3/if HTTP/1.1" 500 128538
ERROR 2023-09-29 12:19:17,702 log 16180 6332 Internal Server Error: /lec3/if
Traceback (most recent call last):
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader_tags.py", line 132, in render
    compiled_parent = self.get_parent(context)
                      ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader_tags.py", line 129, in get_parent
    return self.find_template(parent, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\loader_tags.py", line 107, in find_template
    template, origin = context.template.engine.find_template(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\engine.py", line 161, in find_template
    raise TemplateDoesNotExist(name, tried=tried)
django.template.exceptions.TemplateDoesNotExist: base.html

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\backends\django.py", line 63, in render
    reraise(exc, self.backend)
  File "C:\Users\Koha\FreeTheDjango\Lib\site-packages\django\template\backends\django.py", line 84, in reraise
    raise new from exc
django.template.exceptions.TemplateDoesNotExist: base.html
ERROR 2023-09-29 12:19:17,706 basehttp 16180 6332 "GET /lec3/if HTTP/1.1" 500 128675
INFO 2023-09-29 12:21:20,765 autoreload 9440 14124 Watching for file changes with StatReloader
INFO 2023-09-29 12:21:24,759 basehttp 9440 16008 "GET /lec3/if HTTP/1.1" 200 569
INFO 2023-09-29 12:21:30,032 basehttp 9440 16008 "GET /lec3/for HTTP/1.1" 200 760
INFO 2023-09-29 12:23:54,374 autoreload 9440 14124 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lessonsofdjango\urls.py changed, reloading.
INFO 2023-09-29 12:23:54,998 autoreload 9396 14764 Watching for file changes with StatReloader
INFO 2023-09-29 12:24:05,674 basehttp 9396 4696 "GET / HTTP/1.1" 200 760
INFO 2023-09-29 12:29:11,346 autoreload 9396 14764 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lessonsofdjango\urls.py changed, reloading.
INFO 2023-09-29 12:29:11,943 autoreload 980 6836 Watching for file changes with StatReloader
INFO 2023-09-29 12:29:13,806 basehttp 980 992 "GET / HTTP/1.1" 200 194
INFO 2023-09-29 12:31:05,143 autoreload 980 6836 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\models.py changed, reloading.
INFO 2023-09-29 12:31:05,765 autoreload 15056 13944 Watching for file changes with StatReloader
INFO 2023-09-29 12:31:45,074 autoreload 15056 13944 C:\Users\Koha\Desktop\Учеба GB\FreeTheDjango\lessonsofdjango\lection3app\models.py changed, reloading.
INFO 2023-09-29 12:31:45,712 autoreload 3440 1696 Watching for file changes with StatReloader
INFO 2023-09-29 13:02:43,205 autoreload 15968 15708 Watching for file changes with StatReloader
INFO 2023-09-29 13:02:56,449 basehttp 15968 280 "GET /lec3/author/5 HTTP/1.1" 301 0
WARNING 2023-09-29 13:02:56,461 log 15968 8440 Not Found: /lec3/author/5/
WARNING 2023-09-29 13:02:56,462 basehttp 15968 8440 "GET /lec3/author/5/ HTTP/1.1" 404 4076
INFO 2023-09-29 13:03:57,880 basehttp 15968 8440 "GET /lec3/author/50 HTTP/1.1" 301 0
WARNING 2023-09-29 13:03:57,892 log 15968 15656 Not Found: /lec3/author/50/
WARNING 2023-09-29 13:03:57,892 basehttp 15968 15656 "GET /lec3/author/50/ HTTP/1.1" 404 4079
INFO 2023-09-29 13:04:09,027 basehttp 15968 15656 "GET /lec3/author/23 HTTP/1.1" 301 0
INFO 2023-09-29 13:04:09,041 basehttp 15968 16228 "GET /lec3/author/23/ HTTP/1.1" 200 1281
INFO 2023-09-29 13:04:29,659 basehttp 15968 16228 "GET /lec3/post/230/ HTTP/1.1" 200 582
